pubsub-mapped-api/src/main/java/com/google/pubsub/common/StubCreator.java

```diff
@@ -0,0 +1,38 @@
+package com.google.pubsub.common;
+
+import com.google.pubsub.v1.PublisherGrpc;
+import com.google.pubsub.v1.PublisherGrpc.PublisherBlockingStub;
+import com.google.pubsub.v1.SubscriberGrpc;
+import com.google.pubsub.v1.SubscriberGrpc.SubscriberBlockingStub;
+import com.google.pubsub.v1.SubscriberGrpc.SubscriberFutureStub;
+import java.util.concurrent.TimeUnit;
+
+public class StubCreator {
+
+  private ChannelUtil channelUtil;
+
+  public StubCreator(ChannelUtil channelUtil) {
+    this.channelUtil = channelUtil;
+  }
+
+  public SubscriberBlockingStub getSubscriberBlockingStub() {
+    return SubscriberGrpc.newBlockingStub(channelUtil.getChannel())
+        .withCallCredentials(channelUtil.getCallCredentials());
+  }
+
+  public SubscriberFutureStub getSubscriberFutureStub() {
+    return SubscriberGrpc.newFutureStub(channelUtil.getChannel())
+        .withCallCredentials(channelUtil.getCallCredentials());
+  }
+
+  public SubscriberFutureStub getSubscriberFutureStub(long timeoutMilliseconds) {
+    return SubscriberGrpc.newFutureStub(channelUtil.getChannel())
+        .withCallCredentials(channelUtil.getCallCredentials())
+        .withDeadlineAfter(timeoutMilliseconds, TimeUnit.MILLISECONDS);
+  }
+
+  public PublisherBlockingStub getPublisherBlockingStub() {
+    return PublisherGrpc.newBlockingStub(channelUtil.getChannel())
```

Comment:
As we are using gRPC API directly (the Java Client API does not provide us with all the functionalities that we need), we contacted with gRPC team and they pinged us that creating a channel is expensive (and we should keep one as singleton), but stubs themselves are cheap and can create them on each call if these are reusing the channel.